math/ncr_modulo_p.cpp

```diff
@@ -0,0 +1,138 @@
+/**
+ * @file
+ * @brief This program aims at calculating nCr modulo p
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂